crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -103,7 +103,29 @@ impl Settlement {
                         Internalization::Disable,
                     ),
                     may_revert: boundary.revertable(),
-                }
+                };
+
+                // To prepare rollout, ensure that the domain settlement encoding works and
+                // matches the boundary settlement encoding
+                match encoding::tx(
+                    auction,
+                    &solution,
+                    eth.contracts(),
+                    solution.approvals(eth, Internalization::Enable).await?,
+                    Internalization::Enable,
+                ) {
+                    Ok(domain) => {
+                        if domain.input != tx.internalized.input {
+                            tracing::warn!(
+                                ?domain,
+                                boundary = ?tx.internalized,
+                                "boundary settlement does not match domain settlement"
+                            );
+                        }
+                    }
+                    Err(err) => tracing::warn!(?err, "failed to encode domain settlement"),
```

Comment:
Should an auction/solution ID be specified in the log?

Reply:
Isn't this part of the tracing span already?